Sources/SmithyEventStreamsAuthAPI/MessageDataSigner.swift

```diff
@@ -0,0 +1,36 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import struct Smithy.Attributes
+import struct SmithyEventStreamsAPI.Message
+import struct Foundation.Data
+
+/// Protocol for signing messages.
+public protocol MessageDataSigner {
```

Comment:
Took the event stream-related methods off of `Signer` & split them out to this protocol.